app/controllers/glossary_terms_controller.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+# create and edit Glossary terms
+class GlossaryTermsController < ApplicationController
+  before_action :login_required, except: [
+    :index,
+    :show_past,
+    :show
+  ]
+  before_action :store_location, only: [:index, :show, :destroy]
+  before_action :pass_query_params, only: [:destroy, :show_past]
+
+  # ---------- Actions to Display data (index, show, etc.) ---------------------
+
+  def index
+    @glossary_terms = GlossaryTerm.all.order(:name)
+  end
+
+  def show
+    @glossary_term = GlossaryTerm.find(params[:id].to_s)
+    @canonical_url = glossary_term_url
+    @layout = calc_layout_params
+    @objects = @glossary_term.images
+  end
+
+  # ---------- Actions to Display forms -- (new, edit, etc.) -------------------
+
+  def new
+    @copyright_holder = @user.name
+    @copyright_year = Time.now.utc.year
+    @upload_license_id = @user.license_id
+    @licenses = License.current_names_and_ids(@user.license)
+  end
+
+  def edit
+    @glossary_term = GlossaryTerm.find(params[:id].to_s)
+  end
+
+  # ---------- Actions to Modify data: (create, update, destroy, etc.) ---------
+
+  def create
+    glossary_term = \
+      GlossaryTerm.new(user: @user, name: params[:glossary_term][:name],
+                       description: params[:glossary_term][:description])
+
+    if params[:glossary_term][:upload_image]
+      glossary_term.add_image(process_image(image_args))
+    end
```

Comment:
Fixes an annoying, long-standing (and possibly original) bug:
If a user created a GlossaryTerm without an image, MO would create and save an empty image as the thumb image.
See https://www.pivotaltracker.com/story/show/90725474 and https://www.pivotaltracker.com/story/show/174462200